server/controllers/project.controller/getProjectsForUser.js

```diff
@@ -10,25 +10,44 @@ import { toApi as toApiProjectObject } from '../../domain-objects/Project';
 const createCoreHandler = (mapProjectsToResponse) => async (req, res) => {
   try {
     const { username } = req.params;
+    const currentUser = req.user?._id || '';
+
     if (!username) {
       res.status(422).json({ message: 'Username not provided' });
       return;
     }
+
     const user = await User.findByUsername(username);
     if (!user) {
       res
         .status(404)
         .json({ message: 'User with that username does not exist.' });
       return;
     }
+
     const projects = await Project.find({ user: user._id })
       .sort('-createdAt')
-      .select('name files id createdAt updatedAt')
+      .select('name files id createdAt updatedAt visibility')
       .exec();
-    const response = mapProjectsToResponse(projects);
+
+    const publicProjectsOnly = projects.filter(
+      (project) => project.visibility === 'Public'
+    );
```

Comment:
Ideally we're going to want to do this filtering as part of the Mongoose query, rather than returning all and then filtering the array.

Reply:
Ok so filtering on `visibility: { $eq: 'Public' }` does not seem to give any results, even with `default: 'Public'`.  But we can use a ["not equals" `$ne`](https://www.mongodb.com/docs/manual/reference/operator/query/ne/) to filter for projects which are not private:  `{ $ne: 'Private' }`.

Something like this:
```
const canViewPrivate = req.user && req.user._id === user._id;

const filter = { user: user._id };
if (!canViewPrivate) {
  filter.visibility = { $ne: 'Private' };
}

const projects = await Project.find(filter)
  .sort('-createdAt')
  .select('name files id createdAt updatedAt visibility')
  .exec();
```

Reply:
This is super great, man! The function got reduced by at least 10 lines and it took away the burden from the database as we now don't need to do extra queries for filtering.

Reply:
🤗 Also this sort of setup should work a lot better if we ever get around to implementing pagination.